core/src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -66,8 +66,8 @@
         Setting.positiveTimeSetting("script.cache.expire", TimeValue.timeValueMillis(0), Property.NodeScope);
     public static final Setting<Integer> SCRIPT_MAX_SIZE_IN_BYTES =
         Setting.intSetting("script.max_size_in_bytes", 65535, Property.NodeScope);
-    public static final Setting<Integer> SCRIPT_MAX_COMPILATIONS_PER_MINUTE =
-        Setting.intSetting("script.max_compilations_per_minute", 15, 0, Property.Dynamic, Property.NodeScope);
+    public static final Setting<Integer> SCRIPT_MAX_COMPILATIONS_RATE =
```

Comment:
I'm not sure this new name is better. It has no indication what the base of the rate is? One must find docs to know what a value means, while before it was very clear from the name (per minute). Can we have something to indicate this, or have the values not be a plain integer, but instead a string with some kind of unit? I realize this is creeping the scope a bit, but I think this new setting will just be confusing (eg look at the history of size values without units in 1.x days).

Reply:
I agree, that it should be obvious by reading the setting how it works.

I changed the setting to be a real rate, which consists of an nonnegative integer and a positive timevalue split by a `/` defaulting to `75/5m`. Is this what you were after?

Reply:
Great, exactly what I was hoping for.